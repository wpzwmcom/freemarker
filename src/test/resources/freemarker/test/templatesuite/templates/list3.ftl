[
<#list listables.list as i>
    [<#list listables.list as j>(${i}@${i_index}, ${j}@${j_index})<#sep>, </#sep></#list>]<#sep>,</#sep>
</#list>
]

<#macro hits xs style="">
    <#list xs>
        <p>${xs?size} hits:
        <div class="hits">
            <#switch style>
                <#case "hidden">
                    ...
                    <#break>
                <#case "other">
                    <#items as x>
                        <div class="hitOther">${x}</div>
                    </#items>
                    <#break>
                <#case "none">
                <#default>
                    <#items as x>
                        <div class="hit">${x}</div>
                    </#items>
                    <#break>
            </#switch>
        </div>
    <#else>
        <p>Nothing.
    </#list>
</#macro>

<@hits ['a', 'b'] />

<@hits ['a', 'b'], "other" />

<@hits ['a', 'b'], "hidden" />

<@hits [] />

<#list listables.list as i><#if i_index gt 1>...<#break></#if>${i}<#sep>, </#sep></#list>
<#list listables.list>[<#items as i><#if i_index gt 1>...<#break></#if>${i}<#sep>, </#sep></#items>]</#list>

<@testAutoClosedSep 1..3 />
<@testAutoClosedSep [1] />
<@testAutoClosedSep [] />

<#macro testAutoClosedSep xs>
<#list xs as x>${x}<#sep>, <#else>Empty</#list>
<#list xs as x>${x}<#sep><#if x_index == 0> /*first*/, <#else>, </#if><#else>Empty</#list>
<#list xs>[<#items as x>${x}<#sep>, </#items>]<#else>Empty</#list>
</#macro>