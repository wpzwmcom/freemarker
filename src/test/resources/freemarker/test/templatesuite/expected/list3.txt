[
    [(11@0, 11@0), (11@0, 22@1), (11@0, 33@2)],
    [(22@1, 11@0), (22@1, 22@1), (22@1, 33@2)],
    [(33@2, 11@0), (33@2, 22@1), (33@2, 33@2)]
]


        <p>2 hits:
        <div class="hits">
                        <div class="hit">a</div>
                        <div class="hit">b</div>
        </div>

        <p>2 hits:
        <div class="hits">
                        <div class="hitOther">a</div>
                        <div class="hitOther">b</div>
        </div>

        <p>2 hits:
        <div class="hits">
                    ...
        </div>

        <p>Nothing.

11, 22, ...
[11, 22, ...]

1, 2, 3
1 /*first*/, 2, 3
[1, 2, 3]
1
1
[1]
Empty
Empty
Empty

